packages/remix/src/utils/futureFlags.ts

```diff
@@ -33,3 +40,26 @@ export function getFutureFlagsBrowser(): FutureConfig | undefined {
 export function getFutureFlagsServer(build: ServerBuild): FutureConfig | undefined {
   return build.future;
 }
+
+/**
+ * Read Remix version from module package.json
+ *
+ * @returns The major version number
+ */
+export function getRemixVersionFromPkg(): number | undefined {
```

Comment:
I'm wondering if this causes problems when deploying remix apps to a platform like Vercel. Not sure if we can access and require the package.json files in these runtimes. Did you try this by any chance?

Reply:
> I'm thinking about things like exposed functions/modules or properties on objects.

The release seems to be so smooth, so not really sure there's a visible API change we can use as a foundation between `v2.0.0` and `v1.19.3`. But can dig deeper into this.

> I guess Remix doesn't do us the favour of exporting a version string somewhere 😅

Yes, unfortunately :/

I have just deployed a Remix v1 app to Vercel (Vercel seems to have [it's own issues](https://github.com/remix-run/remix/discussions/7449) deploying Remix v2 at the moment.). I have been able to require `package.json` in a loader, then pass to the client-side. 

https://remix-vercel-kappa.vercel.app/

Reply:
> But can dig deeper into this

I'd say give it a couple of hours. If you find something, great, otherwise let's go with the package.json check. This is still better than users having to manually specify it.

Reply:
[`build.mode`](https://github.com/remix-run/remix/pull/6964) is a good candidate, only introduced in v2 and publicly accessible. If you just check for its presence you should be okay.

Reply:
Thanks @huw, yes it's looking good for this!